localstripe/resources.py

```diff
@@ -244,6 +246,76 @@ def do_expand(path, obj):
         return obj
 
 
+class Balance(object):
+    object = 'balance'
+
+    def __init__(self, **kwargs):
+        if kwargs:
+            raise UserError(400, 'Unexpected ' + ', '.join(kwargs.keys()))
+
+        currencies = ['eur']
+
+        self.object = self.object
+        self.livemode = False
+        self.type = 'balance'
+        self.connect_reserved = []
+        self.available = [
+            {
+                'amount': 2000,
+                'currency': currency,
+                'source_types': {
+                    'card': 2000
+                }
+            }
+            for currency in currencies
+        ]
+        self.pending = [
+            {
+                'amount': 0,
+                'currency': currency,
+                'source_types': {
+                    'card': 0
+                }
+            }
+            for currency in currencies
+        ]
+
+        store[self.object] = self
+
+        schedule_webhook(Event('balance.available', self))
+
+    @classmethod
+    def _api_get(self):
+        obj = store.get(self.object, None)
+        if obj is None:
+            return self()
+        return obj
+
+    def _export(self, expand=None):
+        # expand is passed to be coherent with the server api
+        # but it's not used
+
+        obj = {}
+
+        for key, value in vars(self).items():
+            if not key.startswith('_'):
+                if isinstance(value, StripeObject):
+                    obj[key] = value._export()
+                elif (isinstance(value, list) and len(value) and
+                        isinstance(value[0], StripeObject)):
+                    obj[key] = [item._export() for item in value]
+                elif isinstance(value, dict):
+                    obj[key] = value.copy()
+                else:
+                    obj[key] = value
+
+        return obj
```

Comment:
I think that what Adrien was proposing here https://github.com/adrienverge/localstripe/pull/165#discussion_r555092875 is to use a `Balance._api_retrieve()` direcly in `server.py`, not `api_retrieve()`. Like that no need to define an `_export()` method.
Do you think it's possible?

Reply:
Yes, let's not duplicate code by redefining `_export()` (unless there is really no other way). And we use `_api_retrieve()` (not `_api_get()`) because `GET` are associated to multiple operations (including *retrieve*).

Reply:
Yes, but remember that now it inherits from just `object`... so there are no methods

Reply:
The problem is with the webhook schedule:

![Screenshot 2021-01-13 at 11 00 52](https://user-images.githubusercontent.com/9470158/104437036-9ba60d80-558e-11eb-9884-27a0a1244dd4.png)

Reply:
So we actually _need_ to have the `_export` method defined

Reply:
I agree that, because `extra_apis.append((})` call `func(**data)._export(expand=expand)`, we can't have both:
- untouched `server.py`
- no `Balance._export()` defined

But I think it's possible to have either:
- a simpler `Balance._export()` by stripping all the code that is unnecessary for the `Balance` object.
- or adding a few lines of code inside `server.py`, like:
```py
for cls in (Balance, ):
    for method, url, func in (('GET', '/v1/' + cls.object, cls._api_retrieve), ):
        app.router.add_route(method, url, func(cls, url))
```
With `cls._api_retrieve` defined inside `ressource.py`.

Both solutions work for me (personally I would have tried the second one first).